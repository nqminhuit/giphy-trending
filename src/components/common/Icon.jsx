import PropTypes from "prop-types";
import React from "react";

export default function Icon({ className, iconClass, description }) {

  return (
    <span className={className + " cursor-pointer text-black-50 text-hover-royalblue"}>
      <i className={iconClass}>{description}</i>
    </span>
  );

}

Icon.propTypes = {
  iconClass: PropTypes.string,
  description: PropTypes.string,
};
