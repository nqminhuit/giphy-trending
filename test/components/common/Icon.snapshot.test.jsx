import React from "react";
import renderer from "react-test-renderer";
import Icon from "../../../src/components/common/Icon.jsx";

it("should render <Icon />", () => {
  const component = renderer.create(<Icon
    className="me-3"
    iconClass="bi bi-eye-fill"
    description={1234}
  />).toJSON();
  expect(component).toMatchSnapshot();
});
