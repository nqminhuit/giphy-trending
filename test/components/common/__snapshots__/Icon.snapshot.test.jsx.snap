// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <Icon /> 1`] = `
<span
  className="me-3 cursor-pointer text-black-50 text-hover-royalblue"
>
  <i
    className="bi bi-eye-fill"
  >
    1234
  </i>
</span>
`;
